Occasional race condition in CI/CD pipeline

Run make setup-coverage
📦 Installing gotestsum...
✅ Installed gotestsum
ok  	github.com/bitshepherds/json-schema-manager/cmd/jsm	6.514s	coverage: 34.6% of statements in ./internal/...
ok  	github.com/bitshepherds/json-schema-manager/internal/app	3.217s	coverage: 70.6% of statements in ./internal/...
ok  	github.com/bitshepherds/json-schema-manager/internal/config	1.081s	coverage: 3.4% of statements in ./internal/...
--- FAIL: TestStandardPathResolver (0.00s)
    --- FAIL: TestStandardPathResolver/Abs_returns_absolute_path (0.00s)
        path_resolver_test.go:83: 
            	Error Trace:	/home/runner/work/json-schema-manager/json-schema-manager/internal/fsh/path_resolver_test.go:83
            	Error:      	Received unexpected error:
            	            	getwd: no such file or directory
            	Test:       	TestStandardPathResolver/Abs_returns_absolute_path
FAIL
coverage: 1.2% of statements in ./internal/...
FAIL	github.com/bitshepherds/json-schema-manager/internal/fsh	0.058s
ok  	github.com/bitshepherds/json-schema-manager/internal/repo	1.201s	coverage: 4.0% of statements in ./internal/...
ok  	github.com/bitshepherds/json-schema-manager/internal/report	1.052s	coverage: 5.1% of statements in ./internal/...
ok  	github.com/bitshepherds/json-schema-manager/internal/schema	1.615s	coverage: 63.3% of statements in ./internal/...
ok  	github.com/bitshepherds/json-schema-manager/internal/validator	1.079s	coverage: 0.7% of statements in ./internal/...
?   	github.com/bitshepherds/json-schema-manager/scripts/build	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/clean	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/fmt	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/lint	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/release_check	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/setup	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/snapshot	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/tester	[no test files]
?   	github.com/bitshepherds/json-schema-manager/scripts/version	[no test files]
FAIL
❌ Command failed: exit status 1
exit status 1
make: *** [Makefile:45: test-badge] Error 1
Error: Process completed with exit code 2.


### Root Cause
Two sub-tests run concurrently via `t.Parallel()`:
1. **`Abs returns absolute path`** (line 78) — calls `resolver.Abs("relative/path")`, which internally calls `filepath.Abs()`, which calls `os.Getwd()` to resolve the relative path.
2. **`CanonicalPath fails with deleted CWD`** (line 95) — deliberately calls `os.Chdir(tmp)` and then `os.RemoveAll(tmp)` to delete the process working directory out from under itself.
Because `os.Chdir` mutates **global process state**, when these two sub-tests race, `filepath.Abs` in sub-test #1 can execute *after* the working directory has been deleted by sub-test #2, causing `os.Getwd()` to return `"getwd: no such file or directory"`.
### Why it only fails on Linux
macOS (Darwin) caches the working directory path in the kernel and continues to return the cached value after deletion. Linux performs a real `getcwd` syscall each time, which fails immediately on a deleted directory. The test even has a comment acknowledging this: *"This test might not fail on all Darwin systems due to Getwd caching"*.
### Affected Files
- [internal/fsh/path_resolver_test.go](cci:7://file:///Users/andy/dev/bitshepherds/json-schema-manager/internal/fsh/path_resolver_test.go:0:0-0:0)
### Steps to Reproduce
Run the test suite on Linux. The failure is intermittent due to the inherent race, but reliably occurs in CI.
### Proposed Fix
Remove `t.Parallel()` from the `CanonicalPath fails with deleted CWD` sub-test so it does not run concurrently with other sub-tests that depend on a valid working directory. Since this sub-test mutates global process state (`os.Chdir`), it must be isolated.

--- a/internal/fsh/path_resolver_test.go
+++ b/internal/fsh/path_resolver_test.go
@@ -3,6 +3,7 @@ package fsh
 import (
 	"os"
 	"path/filepath"
+	"runtime"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -92,9 +93,8 @@ func TestStandardPathResolver(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	//nolint:paralleltest // os.Chdir mutates global state
 	t.Run("CanonicalPath fails with deleted CWD", func(t *testing.T) {
-		t.Parallel()
-		// This test might not fail on all Darwin systems due to Getwd caching
 		tmp, err := os.MkdirTemp("", "vanishing-dir")
 		require.NoError(t, err)
 
@@ -109,10 +109,12 @@ func TestStandardPathResolver(t *testing.T) {
 
 		resolver := NewPathResolver()
 		_, err = resolver.CanonicalPath(".")
-		// We don't assert error here because it's flaky on Darwin,
-		// but it's the suggested method.
-		if err != nil {
-			t.Logf("Successfully triggered error: %v", err)
+		if runtime.GOOS == "darwin" {
+			// Darwin caches the working directory path, so Getwd may
+			// succeed even after the directory is removed.
+			t.Logf("darwin: CanonicalPath error (may be nil): %v", err)
+		} else {
+			require.Error(t, err)
 		}
 	})
 

